src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrRegistrations.cs

```diff
@@ -65,7 +66,12 @@ private static void RegisterCommonServices(IServiceCollection services, string g
         services.TryAddTransient<IGitRepoClonerFactory, GitRepoClonerFactory>();
         services.TryAddTransient<VmrCloakedFileScanner>();
         services.TryAddTransient<VmrBinaryFileScanner>();
-        services.AddHttpClient();
+        services.AddHttpClient("GraphQL", httpClient =>
+        {
+            httpClient.BaseAddress = new Uri("https://api.github.com/graphql");
+            httpClient.DefaultRequestHeaders.Add(HeaderNames.Accept, "application/json");
+            httpClient.DefaultRequestHeaders.Add(HeaderNames.UserAgent, "Darc");
+        });
```

Comment:
CodeQL has a rule to ensure that HttpClients are configured to check Certificate Revocation Lists. Unfortunately, HttpClient does not do this by default. There are a few ways to do it, but this may be easiest here. (Also a code search for "CertificateRevocationList" in our repos will find other examples.)

```cs
.ConfigureHttpMessageHandlerBuilder(handler =>
{
    if (handler.PrimaryHandler is HttpClientHandler httpClientHandler)
    {
        httpClientHandler.CheckCertificateRevocationList = true;
    }
    else if (handler.PrimaryHandler is SocketsHttpHandler socketsHttpHandler)
    {
        socketsHttpHandler.SslOptions.CertificateRevocationCheckMode = X509RevocationMode.Online;
    }
    else
    {
        throw new InvalidOperationException($"Could not create client with CRL check, HttpMessageHandler type {handler.PrimaryHandler.GetType().FullName ?? handler.PrimaryHandler.GetType().Name} is unknown.");
    }
});
```

Reply:
thank you, I didn't know about this